kubectl-volsync/cmd/migration_create.go

```diff
@@ -74,24 +123,163 @@ func doMigrationCreate(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+	// Create the empty relationship
 	relation, err := CreateRelationship(configDir, rName, MigrationRelationship)
 	if err != nil {
+		return fmt.Errorf("%w", err)
+	}
+	// Prepare the MigrationParams struct from cmd args
+	migParams, err := prepMigrationParamsStruct(cmd, relation)
+	if err != nil {
+		return fmt.Errorf("failed to build migration params structure from the cmd flags %w", err)
+	}
+	// Check the namespace, if not present already then create the same
+	err = checkAndCreateNamespace(ctx, migParams)
+	if kerrs.IsAlreadyExists(err) {
+		klog.V(2).Info("Namespace: %v already present, proceeding with this namespace", migParams.Dest.Namespace)
+	} else {
 		return err
 	}
 
+	// Check if the pvc already present
+	destPVCExists := false
+	_, err = GetDestinationPVC(ctx, migParams)
+	if err != nil {
+		if len(migParams.Dest.Capacity) == 0 {
+			return fmt.Errorf("%w, please provide the storage capacity to create destination pvc", err)
+		}
+	} else {
+		destPVCExists = true
+	}
+
+	err = CreateDestination(ctx, migParams, destPVCExists)
+	if err != nil {
+		return fmt.Errorf("%w", err)
+	}
+
+	return nil
+}
+
+func CreateDestination(ctx context.Context, migParams *MigrationParams, destPVCExists bool) error {
+	qCap, _ := resource.ParseQuantity(migParams.Dest.Capacity)
+	accessModes := []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}
+	rsyncSpec := &volsyncv1alpha1.ReplicationDestinationRsyncSpec{
+		ReplicationDestinationVolumeOptions: volsyncv1alpha1.ReplicationDestinationVolumeOptions{
+			CopyMethod:  volsyncv1alpha1.CopyMethodNone,
+			Capacity:    &qCap,
+			AccessModes: accessModes,
```

Comment:
Since we are providing `destinationPVC`, we shouldn't provide these here. They should be used to create the PVC (if necessary)